### DISCUSSION TITLE
Problems to plot Rotation curves for the McMillan17 and Irregang13

### DISCUSSION THREAD

**Discussion by edan0314** [jobovy/galpy#541] [Q&A] [answered]
I'm a new user of galpy, I'm trying to validate some references results of orbits integration.
I'm not being able to reproduce the results, and when I was trying to plot the rotation curves for McMillan17 and Irregang13I, it does not plot the curves. The code for this part is:

from galpy.potential import (plotRotcurve, MWPotential as mwp, MWPotential2014 as mwp14)
import matplotlib.pyplot as plt
from galpy.potential.mwpotentials import Irrgang13I as asi
import numpy
from astropy import units
from galpy.util.bovy_coords import XYZ_to_galcencyl, vxvyvz_to_galcencyl, cyl_to_rect
from galpy.orbit import Orbit

plotRotcurve(mwp,Rrange=[0.01,10.],yrange=[0.,1.2],color='r',label='Milk-Way')

plotRotcurve(mwp14,Rrange=[0.01,10.],yrange=[0.,1.2],color='c',linestyle='-',label='Milk-Way 2014', overplot=True)

plotRotcurve(asi,Rrange=[0.01,10.],yrange=[0.,1.2],color='k',linestyle='-',label='Irrgang', overplot=True)

plt.legend()

**Comment by jobovy** [comment, marked as answer]
Do you mean that the plots just don't show up? You generally have to do something to make `matplotlib` show figures in a notebook, e.g.,
```
%matplotlib inline
```
or
```
%pylab inline
```

EDIT: Full answer in thread below, copied here

> I think the issue is one of units. `MWPotential` and `MWPotential2014` by default don't have physical outputs turned on, so they are plotted in internal units. While `McMillan17` and `Irrgang13I` by default return outputs in physical units, so the rotation curve is plotted in units of km/s as a function of kpc. Easiest would be to turn on physical outputs for `MWPotential` and `MWPotential2014`, following the example in the docs (bottom of this section: https://docs.galpy.org/en/latest/reference/potential.html#milky-way-like-potentials). In your case, I think you need
> ```
> plotRotcurve(mwp,Rrange=[0.01,80.],yrange=[0.,270.],color='r',label='Milk-Way',ro=8.,vo=220.)
> plotRotcurve(mwp14,Rrange=[0.01,80.],yrange=[0.,270.],color='c',linestyle='-',label='Milk-Way 2014', overplot=True,ro=8.,vo=220.)
> plotRotcurve(asi,Rrange=[0.01,80.],yrange=[0.,270.],color='k',linestyle='-',label='Irrgang', overplot=True)
> ```

**Reply by edan0314** [reply]
The plot of the McMillan17 and Irregang13I don't show up.
The Milkway and Milkway2014 shows normally

**Reply by jobovy** [reply]
I think the issue is one of units. `MWPotential` and `MWPotential2014` by default don't have physical outputs turned on, so they are plotted in internal units. While `McMillan17` and `Irrgang13I` by default return outputs in physical units, so the rotation curve is plotted in units of km/s as a function of kpc. Easiest would be to turn on physical outputs for `MWPotential` and `MWPotential2014`, following the example in the docs (bottom of this section: https://docs.galpy.org/en/latest/reference/potential.html#milky-way-like-potentials). In your case, I think you need
```
plotRotcurve(mwp,Rrange=[0.01,80.],yrange=[0.,270.],color='r',label='Milk-Way',ro=8.,vo=220.)
plotRotcurve(mwp14,Rrange=[0.01,80.],yrange=[0.,270.],color='c',linestyle='-',label='Milk-Way 2014', overplot=True,ro=8.,vo=220.)
plotRotcurve(asi,Rrange=[0.01,80.],yrange=[0.,270.],color='k',linestyle='-',label='Irrgang', overplot=True)
```

**Reply by edan0314** [reply]
That solved!
Thanks very much!
